配置中心未配置线程池启动报错

## 问题复现

配置中心以及配置文件均未配置动态线程池 ，启动报错：

```java
Caused by: java.lang.NullPointerException: null
	at cn.hippo4j.core.springboot.starter.notify.CoreNotifyConfigBuilder.buildNotify(CoreNotifyConfigBuilder.java:56) ~[classes/:na]
	at cn.hippo4j.message.service.HippoBaseSendMessageService.run(HippoBaseSendMessageService.java:122) ~[classes/:na]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:795) [spring-boot-2.3.2.RELEASE.jar:2.3.2.RELEASE]
```

## 解决方案

作出空判断，为空时打印警告日志，并跳过接下来的流程。

--- a/hippo4j-spring-boot/hippo4j-core-spring-boot-starter/src/main/java/cn/hippo4j/core/springboot/starter/notify/CoreNotifyConfigBuilder.java
+++ b/hippo4j-spring-boot/hippo4j-core-spring-boot-starter/src/main/java/cn/hippo4j/core/springboot/starter/notify/CoreNotifyConfigBuilder.java
@@ -56,7 +56,7 @@ public class CoreNotifyConfigBuilder implements NotifyConfigBuilder {
         boolean globalAlarm = bootstrapCoreProperties.getAlarm();
         List<ExecutorProperties> executors = bootstrapCoreProperties.getExecutors();
         if (CollectionUtil.isEmpty(executors)) {
-            log.error("Failed to buildNotify,executors configuration is empty");
+            log.warn("Failed to build notify, executors configuration is empty.");
             return resultMap;
         }
         List<ExecutorProperties> actual = executors.stream().filter(each -> Optional.ofNullable(each.getNotify()).map(notify -> notify.getIsAlarm()).orElse(false)).collect(Collectors.toList());

--- a/hippo4j-spring-boot/hippo4j-core-spring-boot-starter/src/main/java/cn/hippo4j/core/springboot/starter/notify/CoreNotifyConfigBuilder.java
+++ b/hippo4j-spring-boot/hippo4j-core-spring-boot-starter/src/main/java/cn/hippo4j/core/springboot/starter/notify/CoreNotifyConfigBuilder.java
@@ -29,6 +29,7 @@ import cn.hutool.core.util.StrUtil;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.List;
 import java.util.Map;
@@ -42,6 +43,7 @@ import java.util.stream.Collectors;
  * @date 2022/2/25 00:24
  */
 @AllArgsConstructor
+@Slf4j
 public class CoreNotifyConfigBuilder implements NotifyConfigBuilder {
 
     private final AlarmControlHandler alarmControlHandler;
@@ -53,14 +55,16 @@ public class CoreNotifyConfigBuilder implements NotifyConfigBuilder {
         Map<String, List<NotifyConfigDTO>> resultMap = Maps.newHashMap();
         boolean globalAlarm = bootstrapCoreProperties.getAlarm();
         List<ExecutorProperties> executors = bootstrapCoreProperties.getExecutors();
+        if (CollectionUtil.isEmpty(executors)) {
+            log.error("Failed to buildNotify,executors configuration is empty");
+            return resultMap;
+        }
         List<ExecutorProperties> actual = executors.stream().filter(each -> Optional.ofNullable(each.getNotify()).map(notify -> notify.getIsAlarm()).orElse(false)).collect(Collectors.toList());
         if (!globalAlarm && CollectionUtil.isEmpty(actual)) {
             return resultMap;
         }
-        if (CollectionUtil.isNotEmpty(executors)) {
-            for (ExecutorProperties executor : executors) {
-                resultMap.putAll(buildSingleNotifyConfig(executor));
-            }
+        for (ExecutorProperties executor : executors) {
+            resultMap.putAll(buildSingleNotifyConfig(executor));
         }
         return resultMap;
     }

